### PR TITLE
Remove misleading @Restricted annotation on package

### PR DESCRIPTION
**Pull request by darxriggs**
A package can be annotated. BUT: No restriction is applied!

### CODE REVIEW

**Comment by casz** [comment]
I'd prefer if the API was moved to a separate plugin or we found a way to combine efforts working on an existing API library

**Comment by darxriggs** [comment]
I also opened this issue https://github.com/kohsuke/access-modifier/issues/19.

**Comment by oleg-nenashev** [comment]
+1 for a separate plugin. No opinion about this PR, I am not sure maintainers want to open API

**Comment by oleg-nenashev** [comment]
Ping @casz for the final decision

**Comment by casz** [comment]
🤷‍♂️

**Comment by darxriggs** [comment]
Maybe I missed to add enough information. Seeing your comments I am not sure if the goal of this pull request was fully understood. At least I did not completely understand your comments. So let me try again.

* The _package-info.java_ file was added in 41b4d136ad2efeff27858664b510e30f75333f0b only to provide the `@Restricted(NoExternalUse.class)` annotation on the `com.cloudbees.jenkins.plugins.bitbucket.api` package.
* Doing so has no effect at all.
* It's still possible to use this plugin as a dependency of another plugin and to create an instance of a class inside the mentioned package - for example `new BitbucketException("message")`.
* Therefore I removed _package-info.java_ to get rid of the confusion.

**Comment by casz** [comment]
As I recall this was added to avoid users depending on this plugin to only use the API.
Would be nice to create a API plugin much like 
https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin
https://github.com/jenkinsci/gitlab-api-plugin

**Comment by darxriggs** [comment]
Who is merging it?

**Comment by amuniz** [comment]
> Doing so has no effect at all or none that I am aware of.

I wasn't aware of this issue in the access-modifier-checker. It should be fixed there instead of removing the restriction here.

> That's the whole idea of an api to be the interface that can be used

This package is named `api` because it contains the harness to access Bitbucket's API, not because it is an API itself.

I wouldn't open this for usages by other plugins, cause doing so would make the evolution of this **internal** part of the code much more difficult (would need to keep backward compatibility).

The right thing is to create a `bitbucket-api` plugin as others have already pointed out in this PR.

That's my opinion, but I let @casz to decide as the main maintainer.

**Comment by darxriggs** [comment]
Ok now I understood what this `api` is about.

I added a small Javadoc to the package and annotated all classes in the package to properly restrict them from external use as originally intended.
